app/controllers/sign_up/completions_controller.rb

```diff
@@ -91,6 +91,19 @@ def analytics_attributes(page_occurence)
         in_account_creation_flow: user_session[:in_account_creation_flow] || false,
         needs_completion_screen_reason: needs_completion_screen_reason,
       }
+
+      if page_occurence.present? && DisposableDomain.is_disposable?(email_domain)
+        attributes[:disposable_email_domain] = email_domain
+      end
+
+      attributes
+    end
+
+    def email_domain
+      @email_domain ||= begin
+        email_address = current_user.email_addresses.take.email
+        Mail::Address.new(email_address).domain
+      end
```

Comment:
The ticket was rather ambiguous with the use of "registration", but I'd imagined we'd do this check at the same time we validate the email address, i.e. [`RegisterUserEmailForm`](https://github.com/18F/identity-idp/blob/main/app/forms/register_user_email_form.rb) / [`FormEmailValidator`](https://github.com/18F/identity-idp/blob/main/app/validators/form_email_validator.rb) (called in [`SignUp::RegistrationsController`](https://github.com/18F/identity-idp/blob/main/app/controllers/sign_up/registrations_controller.rb)). That's where we currently check for disposable emails, and where we'd likely enforce the new, larger set of emails in the future. It might also make it a little easier / more accurate to check the specific email at that point in the flow, rather than at the end of account registration.

Reply:
Ah, I thought we wanted to compute statistics on successful registrations before we enforce the ban. And I would expect the analytics attribute to be temporary.

For analysis isn't the emphasis on existing users and successful new registrations?

I can make the change either way.

Reply:
For posterity: @jc-gsa and I [chatted on Slack](https://gsa-tts.slack.com/archives/C05MGJ72GU9/p1702412928155269) and agreed that since our primary concern is to try to identify potentially-legitimate domains being flagged as disposable, the end of account creation is a sensible place to log.